File: Ejercicio 9/profesion.py
from abc import ABC,abstractmethod

class Trabajable(ABC):
    @abstractmethod
    def trabajar(self): pass

class Profesion:
    def __init__(self,nombre):
        self.nombre=nombre

class Medico(Profesion,Trabajable):
    def __init__(self): super().__init__("Medico")
    def trabajar(self): print(self.nombre+" Atendiendo pacientes")

class Ingeniero(Profesion,Trabajable):
    def __init__(self): super().__init__("Ingeniero")
    def trabajar(self): print(self.nombre+" Desarrollando sistema")

class Profesor(Profesion,Trabajable):
    def __init__(self): super().__init__("Profesor")
    def trabajar(self): print(self.nombre+ " Enseñando a estudiantes")

trabajos=[Medico(),Ingeniero(),Profesor()]

for t in trabajos:
    t.trabajar()

File: Ejercicio 9/profesion.java
interface Trabajable{
    void trabajar();
}

abstract class Profesion{
    String nombre;

    Profesion(String n){ nombre=n; }
}

class Medico extends Profesion implements Trabajable{
    Medico(){ super("Medico"); }

    public void trabajar(){
        System.out.println(nombre+" Atendiendo pacientes");
    }
}

class Ingeniero extends Profesion implements Trabajable{
    Ingeniero(){ super("Ingeniero"); }

    public void trabajar(){
        System.out.println( nombre + " Desarrollando sistema");
    }
}

class Profesor extends Profesion implements Trabajable{
    Profesor(){ super("Profesor"); }

    public void trabajar(){
        System.out.println(nombre+ " Enseñando a estudiantes");
    }
}

public class Main{
    public static void main(String[] args){

        Trabajable[] trabajos={
            new Medico(),
            new Ingeniero(),
            new Profesor()
        };

        for(Trabajable t:trabajos)
            t.trabajar();
    }
}
